File: leetcode/11.盛最多水的容器.cpp
/*
 * @lc app=leetcode.cn id=11 lang=cpp
 *
 * [11] 盛最多水的容器
 */

// @lc code=start
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0, r = height.size()-1;
        int ans = 0;
        while (l < r) {
            ans = max(ans, min(height[l], height[r])*(r-l));
            if (height[l]<height[r]) l++;
            else r--;
        }
        return ans;
    }
};
// @lc code=end

File: leetcode/11.盛最多水的容器.py
#
# @lc app=leetcode.cn id=11 lang=python3
#
# [11] 盛最多水的容器
#

# @lc code=start
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r = 0, len(height)-1
        ans = 0
        while (l < r):
            ans = max(ans, (r-l)*min(height[l], height[r]))
            if (height[l] < height[r]):
                l += 1
            else:
                r -= 1
        return ans
        
# @lc code=end

